File: LoggerApp/logger/src/main/java/experiments/com/pixellot/logger/database/LogsDao.java
package experiments.com.pixellot.logger.database;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

/**
 * Created on 18.09.2017.
 */
@Dao
public interface LogsDao {
    @Delete
    void delete(DbLogModel user);

    @Query("SELECT * FROM logs WHERE uid=:id")
    DbLogModel findByName(int id);

    @Query("SELECT * FROM logs")
    List<DbLogModel> getAll();

    @Insert
    void insertAll(DbLogModel... logModels);
}

File: LoggerApp/logger/src/main/java/experiments/com/pixellot/logger/database/LogsDao.kt
package experiments.com.pixellot.logger.database

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Delete
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query

/**
 * Created on 18.09.2017.
 */
@Dao
interface LogsDao {
    @Delete
    fun delete(user: DbLogModel)

    @Query("SELECT * FROM logs WHERE uid=:id")
    fun findByName(id: Int): DbLogModel

    @get:Query("SELECT * FROM logs")
    val all: List<DbLogModel>

    @Insert
    fun insertAll(vararg logModels: DbLogModel)
}
